saferpay_test/src/Form/SaferpayTestForm.php

```diff
@@ -26,10 +28,67 @@ public function getFormId() {
   /**
    * {@inheritdoc}
    */
-  public function buildForm(array $form, FormStateInterface $form_state, TermInterface $taxonomy_term = NULL) {
-    $form['submit'] = array(
+  public function buildForm(array $form, FormStateInterface $form_state, Request $request = NULL) {
+    foreach ($request->query->all() as $key => $value) {
+      drupal_set_message($key . $value);
+    }
+
+    $form_elements = array(
+      'success_link' => $request->query->get('SUCCESSLINK'),
+      'fail_link' => $request->query->get('FAILLINK'),
+      'back_link' => $request->query->get('BACKLINK'),
+      'notify_url' => $request->query->get('NOTIFYURL'),
+      'status' => 'success',
+    );
+
+    $DATA= array(
+      '@MSGTYPE' => '',
+      '@TOKEN' => '',
+      '@VTVERIFY' => '',
+      '@KEYID' => '',
+      '@ID' => '',
+      '@ACCOUNTID' => $request->query->get('ACCOUNTID'),
+      '@PROVIDERID' => '',
+      '@PROVIDERNAME' => '',
+      '@PAYMENTMETHOD' => '',
+      '@ORDERID' => '',
+      '@AMOUNT' => $request->query->get('AMOUNT'),
+      '@CURRENCY' => $request->query->get('CURRENCY'),
+      '@IP' => '',
+      '@IPCOUNTRY' => '',
+      '@CCCOUNTRY' => '',
+      '@MPI_LIABILTYSHIFT' => '',
+      '@MPI_TX_CAVV' => '',
+      '@MPI_XID' => '',
+      '@ECI' => '',
+      '@CAVV' => '',
+      '@XID' => '',
+    );
+
+    foreach ($form_elements as $key => $value) {
+      $form[$key] = array(
+        '#type' => 'hidden',
+        '#value' => $value,
+      );
+    }
+
+    // Don't generate the route, use the submitted url.
```

Comment:
No need to document that we are not doing, but maybe explain a but more what we are doing. (that we are using the submitted success or fail url, based on configuration set by the test...)